ayekan/exporters/snmp-generator/Makefile

```diff
@@ -0,0 +1,356 @@
+
+$(MIBDIR)/ENTITY-STATE-TC-MIB:
+	@echo ">> Downloading Cisco ENTITY-STATE-TC-MIB"
+	@curl $(CURL_OPTS) -o $(MIBDIR)/ENTITY-STATE-TC-MIB "$(CISCO_URL)/ENTITY-STATE-TC-MIB.my"
+
+$(MIBDIR)/ISDN-MIB:
+	@echo ">> Downloading Cisco ISDN-MIB"
+	@curl $(CURL_OPTS) -o $(MIBDIR)/ISDN-MIB "$(CISCO_URL)/ISDN-MIB.my"
+
+$(MIBDIR)/IANA-CHARSET-MIB.txt:
+	@echo ">> Downloading IANA charset MIB"
+	@curl $(CURL_OPTS) -o $(MIBDIR)/IANA-CHARSET-MIB.txt $(IANA_CHARSET_URL)
+
+$(MIBDIR)/IANA-IFTYPE-MIB.txt:
+	@echo ">> Downloading IANA ifType MIB"
+	@curl $(CURL_OPTS) -o $(MIBDIR)/IANA-IFTYPE-MIB.txt $(IANA_IFTYPE_URL)
+
+$(MIBDIR)/IANA-PRINTER-MIB.txt:
+	@echo ">> Downloading IANA printer MIB"
+	@curl $(CURL_OPTS) -o $(MIBDIR)/IANA-PRINTER-MIB.txt $(IANA_PRINTER_URL)
+
+$(MIBDIR)/KEEPALIVED-MIB:
+	@echo ">> Downloading KEEPALIVED-MIB"
+	@curl $(CURL_OPTS) -o $(MIBDIR)/KEEPALIVED-MIB $(KEEPALIVED_URL)
+
+$(MIBDIR)/VRRP-MIB:
+	@echo ">> Downloading VRRP-MIB"
+	@curl $(CURL_OPTS) -o $(MIBDIR)/VRRP-MIB $(VRRP_URL)
+
+$(MIBDIR)/VRRPv3-MIB:
+	@echo ">> Downloading VRRPv3-MIB"
+	@curl $(CURL_OPTS) -o $(MIBDIR)/VRRPv3-MIB $(VRRPV3_URL)
+
+$(MIBDIR)/.kemp-lm:
+	$(eval TMP := $(shell mktemp))
+	@echo ">> Downloading Kemp LM MIBs to $(TMP)"
+	@curl $(CURL_OPTS) -L -o $(TMP) $(KEMP_LM_URL)
+	@unzip -j -d $(MIBDIR) $(TMP) *.txt
+	# Workaround invalid timestamps.
+	@sed -i.bak -E 's/"([0-9]{12})[0-9]{2}Z"/"\1Z"/' $(MIBDIR)/*.RELEASE-B100-MIB.txt
+	@rm $(MIBDIR)/*.RELEASE-B100-MIB.txt.bak
+	@rm -v $(TMP)
+	@touch $(MIBDIR)/.kemp-lm
+
+$(MIBDIR)/MIKROTIK-MIB:
+	@echo ">> Downloading MIKROTIK-MIB"
+	@curl $(CURL_OPTS) -L -o $(MIBDIR)/MIKROTIK-MIB $(MIKROTIK_URL)
+
+$(MIBDIR)/.net-snmp:
+	@echo ">> Downloading NET-SNMP mibs"
+	@curl $(CURL_OPTS) -o $(MIBDIR)/HCNUM-TC $(NET_SNMP_URL)/HCNUM-TC.txt
+	@curl $(CURL_OPTS) -o $(MIBDIR)/HOST-RESOURCES-MIB $(NET_SNMP_URL)/HOST-RESOURCES-MIB.txt
+	@curl $(CURL_OPTS) -o $(MIBDIR)/IF-MIB $(NET_SNMP_URL)/IF-MIB.txt
+	@curl $(CURL_OPTS) -o $(MIBDIR)/IP-MIB $(NET_SNMP_URL)/IP-MIB.txt
+	@curl $(CURL_OPTS) -o $(MIBDIR)/INET-ADDRESS-MIB $(NET_SNMP_URL)/INET-ADDRESS-MIB.txt
+	@curl $(CURL_OPTS) -o $(MIBDIR)/IPV6-TC $(NET_SNMP_URL)/IPV6-TC.txt
+	@curl $(CURL_OPTS) -o $(MIBDIR)/NET-SNMP-MIB $(NET_SNMP_URL)/NET-SNMP-MIB.txt
+	@curl $(CURL_OPTS) -o $(MIBDIR)/NET-SNMP-TC $(NET_SNMP_URL)/NET-SNMP-TC.txt
+	@curl $(CURL_OPTS) -o $(MIBDIR)/SNMP-FRAMEWORK-MIB $(NET_SNMP_URL)/SNMP-FRAMEWORK-MIB.txt
+	@curl $(CURL_OPTS) -o $(MIBDIR)/SNMPv2-MIB $(NET_SNMP_URL)/SNMPv2-MIB.txt
+	@curl $(CURL_OPTS) -o $(MIBDIR)/SNMPv2-SMI $(NET_SNMP_URL)/SNMPv2-SMI.txt
+	@curl $(CURL_OPTS) -o $(MIBDIR)/SNMPv2-TC $(NET_SNMP_URL)/SNMPv2-TC.txt
+	@curl $(CURL_OPTS) -o $(MIBDIR)/UCD-SNMP-MIB $(NET_SNMP_URL)/UCD-SNMP-MIB.txt
+	@touch $(MIBDIR)/.net-snmp
+
+$(MIBDIR)/PICO-IPSEC-FLOW-MONITOR-MIB.txt:
+	@echo ">> Downloading PICO-IPSEC-FLOW-MONITOR-MIB.txt"
+	@curl $(CURL_OPTS) -o $(MIBDIR)/PICO-IPSEC-FLOW-MONITOR-MIB.txt "$(NEC_URL)/PICO-IPSEC-FLOW-MONITOR-MIB.txt"
```

Comment:
It looks like this server doesn't like the curl user agent string:
```
curl -L -sS --retry 3 --retry-delay 3 --fail -o mibs/PICO-IPSEC-FLOW-MONITOR-MIB.txt "https://jpn.nec.com/univerge/ix/Manual/MIB/PICO-IPSEC-FLOW-MONITOR-MIB.txt"
curl: (22) The requested URL returned error: 403
```
It works from firefox and it works if I change the u-a string. E.g.
```bash
$ curl -A"I AM BATMAN" -L -sS --retry 3 --retry-delay 3 --fail -o mibs/PICO-IPSEC-FLOW-MONITOR-MIB.txt "https://jpn.nec.com/univerge/ix/Manual/MIB/PICO-IPSEC-FLOW-MONITOR-MIB.txt"
```